### PR TITLE
created autocomment-pr-merge.yml

### LINKED ISSUE #120: [Feature]: Auto Commenting feature for PR Merged
### Description

**As current workflow files only autocomments on issues created and PR raised**,

This feature aims to solve the problem of inconsistent and delayed communication following the merging of PRs. By automating the commenting process, it ensures that contributors receive timely feedback and acknowledgments. It also reduces the administrative burden on maintainers, allowing them to focus more on code reviews and other critical tasks.

### Screenshots

PR merged

### Checklist

- [X] I have checked the existing [issues](https://github.com/pooranjoyb/popShop/issues)
- [X] I have read the [Contributing Guidelines](https://github.com/pooranjoyb/popShop/blob/main/CONTRIBUTING.md)
- [X] I am willing to work on this issue (optional)

### PR DESCRIPTION
**Pull request by Asymtode712**
This feature aims to solve the problem of inconsistent and delayed communication following the merging of PRs. By automating the commenting process, it ensures that contributors receive timely feedback and acknowledgments. It also reduces the administrative burden on maintainers, allowing them to focus more on code reviews and other critical tasks.

Fixes #120 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Refactor Code
- [ ] A documentation update
- [ ] Others(mentioned in the issue number)

# Screenshort

PR merged

# Checklist:
give a X on it which is applicable

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by pooranjoyb — COMMENTED** [review]
Great job, @Asymtode712! 🎉 Thank you for submitting your pull request. Your contribution is valuable and we appreciate your efforts to improve our project.<p>We will promptly review your changes and offer feedback. Keep up the excellent work! Kindly remember to check our [contributing guidelines](https://github.com/pooranjoyb/popShop/blob/dev/CONTRIBUTING.md)</p>

**Comment by Asymtode712** [comment]
@pooranjoyb please review my PR and merge it if it is good to go

File: .github/workflows/autocomment-pr-merge.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      pull-requests: write
+    if: github.event.pull_request.merged == true
+
+    steps:
+    - name: Checkout Repository
+      uses: actions/checkout@v2
+
+    - name: Add Comment to Issue
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (pooranjoyb) [inline .github/workflows/autocomment-pr-merge.yml:24]: change this to `USER_TOKEN`

File: .github/workflows/autocomment-pr-merge.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      run: |
+        COMMENT=$(cat <<EOF
+        {
+          "body": "🎉 Your pull request has been successfully merged! 🎉 Thank you for your valuable contribution to our project. Your efforts are greatly appreciated. Feel free to reach out if you have any more contributions or if there's anything else we can assist you with. Keep up the fantastic work! 🚀"
+        }
+        EOF
+        )
+        curl -X POST \
+          -H "Authorization: Bearer $GITHUB_TOKEN" \
```
Comment (pooranjoyb) [inline .github/workflows/autocomment-pr-merge.yml:33]: change this to `USER_TOKEN`

**Review by pooranjoyb — CHANGES_REQUESTED** [review]

File: .github/workflows/autocomment-pr-merge.yml (hunk 1 above)
Comment (Asymtode712) [reply .github/workflows/autocomment-pr-merge.yml:24]: ok

File: .github/workflows/autocomment-pr-merge.yml (hunk 2 above)
Comment (Asymtode712) [reply .github/workflows/autocomment-pr-merge.yml:33]: ok

**Comment by Asymtode712** [comment]
@pooranjoyb done with the changes as requested by you

**Comment by pooranjoyb** [comment]
Bruh the key should remain the same as  `GITHUB_TOKEN` 
I said only the values to `USER_TOKEN`
@Asymtode712

**Comment by Asymtode712** [comment]
@pooranjoyb you have assign your PAT to USER_TOKEN right?

**Comment by Asymtode712** [comment]
@pooranjoyb If yes then it wouldn't affect the code

**Comment by pooranjoyb** [comment]
Its already set. Just make it `GITHUB_TOKEN : USER_TOKEN` 
@Asymtode712

**Comment by Asymtode712** [comment]
@pooranjoyb ok as you say but it won't make any difference as the key is just a mere variable so it can be named anything the thing whcih matters is the value

**Comment by Asymtode712** [comment]
@pooranjoyb done the changes, please review and merge it

**Comment by pooranjoyb** [comment]
Nice work @Asymtode712 :rocket:

**Pull request merged**